homeassistant/components/automation/state.py

```diff
@@ -73,16 +73,13 @@ def state_automation_listener(event: Event):
 
         from_s = event.data.get("old_state")
         to_s = event.data.get("new_state")
+        old_state = getattr(from_s, "state", None)
+        new_state = getattr(to_s, "state", None)
 
         if (
-            (from_s is not None and not match_from_state(from_s.state))
-            or (to_s is not None and not match_to_state(to_s.state))
-            or (
-                not match_all
-                and from_s is not None
-                and to_s is not None
-                and from_s.state == to_s.state
-            )
+            not match_from_state(old_state)
+            or not match_to_state(new_state)
+            or (not match_all and old_state == new_state)
```

Comment:
As best I can tell this looks right now. It also seems that lines 104-111 below are now redundant and can be deleted.

Reply:
you're totally right, removed on https://github.com/home-assistant/core/pull/38032/commits/832eda5c791cbc442adfcac292846397c9dc1740 :)